app/Http/Controllers/ExportController.php

```diff
@@ -90,32 +94,34 @@ public function devices(Request $request, $idevents = NULL, $idgroups = NULL)
             fputcsv($file, $columns);
 
             foreach ($all_devices as $device) {
-                $wasteImpact = 0;
-                $co2Diverted = 0;
-
-                if ($device->isFixed()) {
-                    if ($device->deviceCategory->powered) {
-                        $wasteImpact = $device->eWasteDiverted();
-                        $co2Diverted = $device->eCo2Diverted($eEmissionRatio, $displacementFactor);
-                    } else {
-                        $wasteImpact = $device->uWasteDiverted();
-                        $co2Diverted = $device->uCo2Diverted($uEmissionratio, $displacementFactor);
+                if (User::userCanSeeEvent($me, $event)) {
```

Comment:
I think ExportController::devices is called from https://therestartproject.org/download-dataset - a public call with no user as far as I know.  So wonder if this would then mean it returns no data?

Reply:
I wouldn't have thought so.  `userCanSeeEvent` handles the case where the user is logged out (i.e. null) and would return true if the event and group have been approved.  But consider.

Reply:
Yes perfect, I realised this when I got to the userCanSeeEvent method - if event and group both approved then the event/item data will be visible.